### PR TITLE
open links in main process too

### LINKED ISSUE #1945: Opening Links with Windows Desktop in Full Screen mode
In Windows, Desktop opens the following links in IE, but behind the scenes leaving the novice user somewhat clueless to the fact that it was opened in a browser. 

`Repo > View on GH`
`Branch > Compare on GH`
`See what's new`

Other links like `Help > Report an Issue` are opened in IE, but surfaced in front of Desktop. Perhaps we should be doing the same for all links?

Mac OS had no issues.

v0.5.9, Windows 10

### PR DESCRIPTION
**Pull request by shiftkey**
Fixes #1945 by moving these to IPC calls.

### CODE REVIEW

File: app/src/lib/dispatcher/app-shell.ts [hunk 1]
```diff
@@ -3,15 +3,23 @@ import { shell as electronShell, ipcRenderer } from 'electron'
...
   readonly showItemInFolder: (path: string) => void
 }
 
 export const shell: IAppShell = {
   moveItemToTrash: electronShell.moveItemToTrash,
   beep: electronShell.beep,
-  openExternal: electronShell.openExternal,
+  openExternal: (path) => {
+    return new Promise<boolean>((resolve, reject) => {
+      ipcRenderer.once('open-external-result', (event: Electron.IpcMessageEvent, { result }: { result: boolean} ) => {
```
Comment (joshaber) [inline app/src/lib/dispatcher/app-shell.ts]: Style nit: space after `boolean`

Comment (joshaber) [inline app/src/lib/dispatcher/app-shell.ts]: Nevermind, not worth it given that Prettier is so close.

File: app/src/main-process/main.ts [hunk 2]
```diff
@@ -276,6 +276,11 @@ app.on('ready', () => {
     reportError(error, extra)
   })
 
+  ipcMain.on('open-external', (event: Electron.IpcMessageEvent, { id, path }: { id: number, path: string }) => {
```
Comment (joshaber) [inline app/src/main-process/main.ts]: It doesn't look like we use `id` or provide it. Can we 🔥 it?

Comment (shiftkey) [inline app/src/main-process/main.ts]: I thought I'd need to track the IPC request but that's unnecessary :fire::fire::fire:

**Comment by joshaber** [comment]
This looks good but has some conflicts 🙈

**Pull request merged**